### PR TITLE
aws_request_signing: Credential file home directory fix

### PR DESCRIPTION
**Pull request by nbaws**
Commit Message: Credential file home directory fix
Additional Description: Minor change to resolve home directory path correctly
Risk Level: Low
Testing: Unit
Docs Changes: Changelog bugfix
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by nbaws** [comment]
@suniltheta

File: changelogs/current.yaml
Comment (suniltheta) [inline changelogs/current.yaml:138]: this can be just `aws` and not `aws_request_signer`

File: source/extensions/common/aws/credentials_provider_impl.cc
Comment (suniltheta) [inline source/extensions/common/aws/credentials_provider_impl.cc:212]: Do we want to move this below L214 so that in the same debug log we can also capture the extracted profile info?

File: source/extensions/common/aws/credentials_provider_impl.cc
Comment (suniltheta) [inline source/extensions/common/aws/credentials_provider_impl.cc:208]: Is ot possible to utilize the func getEnvironmentVariableOrDefault like below?
```
const auto home = getEnvironmentVariableOrDefault("HOME", "");
const auto default_credentials_file_path = absl::StrCat(home, DEFAULT_AWS_SHARED_CREDENTIALS_FILE);
```

File: test/extensions/common/aws/credentials_provider_impl_test.cc
Comment (suniltheta) [inline test/extensions/common/aws/credentials_provider_impl_test.cc:18]: are these unused?

File: test/extensions/common/aws/credentials_provider_impl_test.cc
Comment (suniltheta) [inline test/extensions/common/aws/credentials_provider_impl_test.cc:140]: I believe this logger is not used?

**Review by suniltheta — CHANGES_REQUESTED** [review]

File: test/extensions/common/aws/credentials_provider_impl_test.cc
Comment (nbaws) [reply test/extensions/common/aws/credentials_provider_impl_test.cc:140]: correct. resolved

File: test/extensions/common/aws/credentials_provider_impl_test.cc
Comment (nbaws) [reply test/extensions/common/aws/credentials_provider_impl_test.cc:18]: this was flagging as follows:
`Using decl 'MockMetadataFetcher' is unused (fix available)`

Envoy::Extensions::Common::Aws::MockMetadataFetcher class is already defined via 

#include "test/extensions/common/aws/mocks.h"

File: source/extensions/common/aws/credentials_provider_impl.cc
Comment (nbaws) [reply source/extensions/common/aws/credentials_provider_impl.cc:208]: yah this is a nicer way. implemented.

File: source/extensions/common/aws/credentials_provider_impl.cc
Comment (nbaws) [reply source/extensions/common/aws/credentials_provider_impl.cc:212]: done

**Review by suniltheta — APPROVED** [review]
thanks for fixing this.

File: source/extensions/common/aws/credentials_provider_impl.cc
Comment (adisuissa) [inline source/extensions/common/aws/credentials_provider_impl.cc:201]: FWIW, this seems like a behavior change - An upgrade to this Envoy may change the used credentials file. I'm not sure if it is intended or not, and will let the codeowner @suniltheta decide.
I suggest thinking about moving it to the end, so it won't take precedence over AWS_SHARED_CREDENTIALS_FILE.

**Review by adisuissa — COMMENTED** [review]
Thanks for your contribution!
Left a few small comments, but will let @suniltheta drive this review as codeowner.

File: source/extensions/common/aws/credentials_provider_impl.cc
Comment (suniltheta) [reply source/extensions/common/aws/credentials_provider_impl.cc:201]: I believe this change won't make `default_credentials_file_path` take precedence over `AWS_SHARED_CREDENTIALS_FILE`. Env `AWS_SHARED_CREDENTIALS_FILE` is read first on L206 before falling back to using `default_credentials_file_path` as default. Please let me know if I missed something.

For existing users who are upgrading Envoy, the Envoy will look for file in `$HOME/.aws/credentials` instead of `~/.aws/credentials` as a default location, which I think should be ok.

checking if @michaelfinch (original author) has any suggestions.

Comment (nbaws) [reply source/extensions/common/aws/credentials_provider_impl.cc:201]: ty for your feedback :) the default location was never working, as `~` is not expanded by Envoy (unless the user had actually created a directory called `~`). This fix was to address that expansion as originally intended but not expected to change behaviour.

**Review by adisuissa — DISMISSED** [review]
LGTM, thanks!

Merging as @suniltheta (codeowner) approved these changes.

**Comment by adisuissa** [comment]
Oops, needs to merge main.

File: test/extensions/common/aws/credentials_provider_impl_test.cc
Comment (suniltheta) [reply test/extensions/common/aws/credentials_provider_impl_test.cc:161]: @nbaws do you want to remove these unrelated changes before merging?

**Review by suniltheta — APPROVED** [review]

File: changelogs/current.yaml
Comment (phlax) [inline changelogs/current.yaml:165]:
```suggestion
    Default credentials file location was relying on incorrect ``~`` expansion. This fix now resolves to the correct path
    by using ``HOME`` environment variable, if the credentials file location is not explicitly specified.
```

**Review by phlax — COMMENTED** [review]
@nbaws thanks for working on this - some tiny docs nits and then i think this should be good to land

**Review by phlax — APPROVED** [review]
lgtm, thanks @nbaws

**Pull request merged**
